### DISCUSSION TITLE
CsvWriter incomplete lines at end of file

### DISCUSSION THREAD

**Discussion by AOONIH** [bonsai-rx/bonsai#1396] [Q&A]
I am saving outputs from the spinnaker capture node with CsvWriter and the frames with VideoWriter. I'm getting fewer lines with than frames and often the last line is incomplete. I'm making sure to delay for a few seconds (~5s) after the last frame has been captured but even with that I still this issue.
![image](https://github.com/bonsai-rx/bonsai/assets/40366898/13b7c09f-cba0-4576-b266-f37b77524c42)
[DO69_230529_eye0_timestamps.csv](https://github.com/bonsai-rx/bonsai/files/11598877/DO69_230529_eye0_timestamps.csv)
[DO70_230529_eye0_timestamps.csv](https://github.com/bonsai-rx/bonsai/files/11598880/DO70_230529_eye0_timestamps.csv)

**Comment by bruno-f-cruz** [comment]
Heya, can you provide the shortest workflow that reproduces the issue? thanks!
